tests/kernel/mem_protect/mem_protect/src/mem_domain.c

```diff
@@ -331,3 +331,34 @@ void test_mem_domain_migration(void)
 
 	k_thread_join(&child_thread, K_FOREVER);
 }
+
+
+K_MEM_PARTITION_DEFINE(app_part, ro_buf, sizeof(ro_buf),
+		       K_MEM_PARTITION_P_RW_U_RW);
+
+/**
+ * @brief Test system assert when new partition overlaps existing partition
+ *
+ * When adding the new partition to a memory domain the system must assert that
+ * it does not overlap with any other existing partitions in the domain.
+ * Test to add new partition which has same start address as an existing one,
+ * after that must happen an assertion error indicating that new partition
+ * overlaps existing one.
+ *
+ * @ingroup kernel_memprotect_tests
+ *
+ * @see k_mem_domain_add_partition()
+ */
+void test_mem_part_overlap(void)
+{
+	/* Before beginning that test there is already partition ro_part
+	 * in the memory domain. Memory domain has one free slot for one more new
+	 * partition. We must keep that initial condition.
+	 */
+
+	set_fault_valid(true);
+
+	app_part.start = ro_part.start;
+
+	k_mem_domain_add_partition(&test_domain, &app_part);
+}
```

Comment:
An assertion will be seen. Does not it mean sanitycheck will fail?

Reply:
It won't cause sanitycheck failure if test passed. Because we are using function `set_fault_valid()` with flag `true`. That means expected behavior of the test is a failure.